extensions/spring-xd-extension-batch/src/main/java/org/springframework/batch/step/tasklet/x/AbstractProcessBuilderTasklet.java

```diff
@@ -73,25 +88,71 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
 
 		String commandClassPath = createClassPath(this.getClass());
 
-		ProcessBuilder pb = new ProcessBuilder(command).redirectErrorStream(true);
+		File out = File.createTempFile(commandName + "-", ".out");
+		stepExecution.getExecutionContext().putString(commandName.toLowerCase() + ".system.out", out.getAbsolutePath());
+		logger.info(commandName + " system.out: " + out.getAbsolutePath());
+		File err = File.createTempFile(commandName + "-", ".err");
```

Comment:
We could consider this as an enhancement down the road, how would this be specified though? We could name them like xd-sqoop-job1-step0.out and put them in XD's log directory.